src/main/scala/viper/server/vsi/VerificationServer.scala

```diff
@@ -76,196 +38,167 @@ class JobPool(val MAX_ACTIVE_JOBS: Int = 3) {
   *  between a VerificationTask's backend and the server. The Terminator Actor is in charge of
   *  terminating both individual processes and the server.
   */
-trait VerificationServer extends Unpacker {
+trait VerificationServer extends Post {
+
+  type AST
 
   implicit val system: ActorSystem = ActorSystem("Main")
-  implicit val executionContext = ExecutionContext.global
+  implicit val executionContext: ExecutionContextExecutor = ExecutionContext.global
   implicit val materializer: ActorMaterializer = ActorMaterializer()
 
-  protected var jobs: JobPool = _
+  protected var _termActor: ActorRef = _
+
+  implicit val ast_id_fact: Int => AstJobId = AstJobId.apply
+  implicit val ver_id_fact: Int => VerJobId = VerJobId.apply
+
+  protected var ast_jobs: JobPool[AstJobId, AstHandle[AST]] = _
+  protected var ver_jobs: JobPool[VerJobId, VerHandle] = _
+
   var isRunning: Boolean = false
 
-  /** Configures an instance of ViperCoreServer.
+  /** Configures an instance of VerificationServer.
     *
     * This function must be called before any other. Calling any other function before this one
     * will result in an IllegalStateException.
     * */
   def start(active_jobs: Int): Unit = {
-    jobs = new JobPool(active_jobs)
-    _termActor = system.actorOf(Terminator.props(), "terminator")
+    ast_jobs = new JobPool("Http-AST-pool", active_jobs)
```

Comment:
this seems to be at least partially redundant to `VerificationServerHttp`. Why is this?

Reply:
This method overrides ```start``` of ```VerificationServer```. I've added the ```override``` keyword to make this more apparent. Maybe there's a more elegant way to do this by calling super.start(), but for some reason, I cannot see it.